### DISCUSSION TITLE
Enabling USB Video Input?

### DISCUSSION THREAD

**Discussion by tmdarwen** [OE4T/meta-tegra#884] [Q&A] [answered]
I'm using an [Avermedia ExtremeCap UVC - BU110](https://www.avermedia.com/us/product-detail/BU110) device to input video into the Jetson.  When using the [NVIDIA Jetson Linux](https://docs.nvidia.com/jetson/l4t/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/quick_start.html#wwpID0E0BE0HA) image I simply plug the device into a USB 3.0 slot on my Jetson Mate and it will show up as a new video device in `/dev` and then I can use it to process video input.

This isn't happening with my meta-tegra image.  When I plug in the Avermedia device no new devices show up in `/dev`.   I imagine I'm missing some sort of USB UVC Linux driver on my meta-tegra image.  Very possibly [this](https://www.kernel.org/doc/html/v4.9/media/v4l-drivers/uvcvideo.html).  I'm not sure how to get this built into my meta-tegra image though.

I grepped the `repos` dir of `tegra-demo-distro` for uvc and only found it mentioned in these four files:
```
./meta-openembedded/meta-networking/recipes-protocols/freediameter/files/0001-Fix-testcnx-expired-CA-data.patch
./meta-tegra/contrib/recipes-multimedia/gstreamer-1.14/gstreamer1.0-plugins-bad_1.14.4.bb
./poky/meta/recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.16.3.bb
./meta-mender-community/meta-mender-sunxi/recipes-kernel/linux/linux-friendlyarm/defconfig
```

None of those appear to be what I'm looking for.

Any idea if there's a layer I can add to get this driver in my meta-tegra image?

**Comment by madisongh** [comment, marked as answer]
Just add `kernel-module-uvcvideo` to your image, or `kernel-modules` if you want all of the kernel modules (like NVIDIA's Ubuntu does).  Typically for an embedded systems build you don't want that, though.

**Reply by tmdarwen** [reply]
Awesome!  Adding `kernel-module-uvcvideo` worked!  Thanks again!
